components/butterfly/src/server/mod.rs

```diff
@@ -1168,16 +1169,34 @@ impl fmt::Display for Server {
 }
 
 fn persist_loop(server: &Server) {
-    // TODO: Make this configurable with EnvConfig. That trait needs to move
-    // to common or core first
-    const MIN_LOOP_PERIOD: Duration = Duration::from_secs(30);
+    struct PersistLoopPeriod(Duration);
+
+    impl EnvConfig for PersistLoopPeriod {
+        const ENVVAR: &'static str = "HAB_PERSIST_LOOP_PERIOD_SECS";
+    }
+
+    impl Default for PersistLoopPeriod {
+        fn default() -> Self { Self(Duration::from_secs(30)) }
+    }
+
+    impl std::str::FromStr for PersistLoopPeriod {
+        type Err = std::num::ParseIntError;
+
+        fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+            Ok(Self(Duration::from_secs(s.parse()?)))
+        }
+    }
 
     loop {
+        habitat_common::sync::mark_thread_alive();
+
         let before_persist = Instant::now();
         server.persist_data();
         let time_to_persist = before_persist.elapsed();
         trace!("persist_data took {:?}", time_to_persist);
-        match MIN_LOOP_PERIOD.checked_sub(time_to_persist) {
+        match PersistLoopPeriod::configured_value().0
+                                                   .checked_sub(time_to_persist)
+        {
```

Comment:
Given that this will never change once the binary has started, why not grab the configured value outside of the loop, so you only create it once?

Reply:
Good point; it was originally a constant during my development and I added in the `EnvConfig` later. Will change.